speechbrain/lobes/models/huggingface_transformers/huggingface.py

```diff
@@ -0,0 +1,372 @@
+        is: model.wav2vec2.weight.data (wav2vec2 must be removed before loading).
+        """
+        if modify_state_dict_partial_fn is not None:
+            modified_state_dict = modify_state_dict_partial_fn(path)
+        else:
+            modified_state_dict = torch.load(path, map_location="cpu")
+
+        incompatible_keys = self.model.load_state_dict(
+            modified_state_dict, strict=False
+        )
+        for missing_key in incompatible_keys.missing_keys:
+            logger.warning(
+                f"During parameter transfer to {self.model} loading from "
+                + f"{path}, the transferred parameters did not have "
+                + f"parameters for the key: {missing_key}"
+            )
+        for unexpected_key in incompatible_keys.unexpected_keys:
+            logger.warning(
+                f"The param with the key: {unexpected_key} is discarded as it "
+                + "is useless for finetuning this HuggingFaceTransformer."
+            )
+
+    def forward(self, **kwargs):
+        """Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def forward_encoder(self, **kwargs):
+        """Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def forward_decoder(self, **kwargs):
+        """Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def decode(self, **kwargs):
+        """Might be useful for models like mbart, which can exploit SB's beamsearch for inference
+        Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def encode(self, **kwargs):
+        """Customed encoding for inference
+        Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def freeze_model(self, model):
+        """
+        Freezes parameters of a model.
+        This should be overrided too, depending on users' needs, for example, adapters use.
+
+        Arguments
+        ---------
+        model : from AutoModel.from_config
+            Valid HuggingFace transformers model object.
+        """
+        model.eval()
+        for param in model.parameters():
+            param.requires_grad = False
+
+    def override_config(self, config):
+        """Users should modify this function according to their own tasks."""
+        return config
+
+    def load_feature_extractor(self, source, cache_dir, **kwarg):
+        """Load model's feature_extractor from the hub"""
+        self.feature_extractor = AutoFeatureExtractor.from_pretrained(
+            source, cache_dir=cache_dir, **kwarg
+        )
+
+    def load_tokenizer(self, source, **kwarg):
+        """Load model's tokenizer from the hub"""
+        self.tokenizer = AutoTokenizer.from_pretrained(source, **kwarg)
+
+
+def make_padding_masks(src, wav_len=None, pad_idx=0):
```

Comment:
Did you double-check that we don't have this somewhere already? I really think we do.